### PR TITLE
Add polarity axioms in CegarValues

### PR DESCRIPTION
**Pull request by makaimann**

### CODE REVIEW

File: engines/cegar_values.cpp [hunk 1]
```diff
@@ -105,13 +107,29 @@ class ValueAbstractor : public smt::IdentityWalker
           return Walker_Continue;
         }
 
+        fresh_solver_->push();
```
Comment (ahmed-irfan) [inline engines/cegar_values.cpp:110]: this seems like a hack to check if a value is positive or negative. Right? If so, please add a comment to improve this in future.

File: engines/cegar_values.cpp [hunk 2]
```diff
@@ -269,6 +288,11 @@ void CegarValues<Prover_T>::cegar_abstract()
   super::bad_ = va.visit(super::bad_);
   cegval_bad_ = to_cegval_solver_.transfer_term(super::bad_, BOOL);
 
+  // add polarity axioms from abstraction
+  for (const auto & ax : va.polarity_axioms()) {
+    prover_ts_.add_constraint(ax);
```
Comment (ahmed-irfan) [inline engines/cegar_values.cpp:293]: is it possible to add this to trans only?

Comment (makaimann) [reply engines/cegar_values.cpp]: Good idea. Added here 266b7d06c797951eba0cf3706ffc6f8fb430b57e. Does that look all right?

File: engines/cegar_values.cpp (hunk 1 above)
Comment (makaimann) [reply engines/cegar_values.cpp:113]: Yeah added with 1d86cbb257beb95ab3b712739dc17bc20773633d

**Review by ahmed-irfan — APPROVED** [review]

**Pull request merged**